File: web/src/api/bbPiPerson.js
import service from '@/utils/request'

// @Tags BbPiPerson
// @Summary 创建BbPiPerson
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiPerson true "创建BbPiPerson"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiPerson/createBbPiPerson [post]
export const createBbPiPerson = (data) => {
  return service({
    url: '/bbPiPerson/createBbPiPerson',
    method: 'post',
    data
  })
}

// @Tags BbPiPerson
// @Summary 删除BbPiPerson
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiPerson true "删除BbPiPerson"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /bbPiPerson/deleteBbPiPerson [delete]
export const deleteBbPiPerson = (data) => {
  return service({
    url: '/bbPiPerson/deleteBbPiPerson',
    method: 'delete',
    data
  })
}

// @Tags BbPiPerson
// @Summary 删除BbPiPerson
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除BbPiPerson"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /bbPiPerson/deleteBbPiPerson [delete]
export const deleteBbPiPersonByIds = (data) => {
  return service({
    url: '/bbPiPerson/deleteBbPiPersonByIds',
    method: 'delete',
    data
  })
}

// @Tags BbPiPerson
// @Summary 更新BbPiPerson
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiPerson true "更新BbPiPerson"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /bbPiPerson/updateBbPiPerson [put]
export const updateBbPiPerson = (data) => {
  return service({
    url: '/bbPiPerson/updateBbPiPerson',
    method: 'put',
    data
  })
}

// @Tags BbPiPerson
// @Summary 用id查询BbPiPerson
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.BbPiPerson true "用id查询BbPiPerson"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /bbPiPerson/findBbPiPerson [get]
export const findBbPiPerson = (params) => {
  return service({
    url: '/bbPiPerson/findBbPiPerson',
    method: 'get',
    params
  })
}

// @Tags BbPiPerson
// @Summary 分页获取BbPiPerson列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取BbPiPerson列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiPerson/getBbPiPersonList [get]
export const getBbPiPersonList = (params) => {
  return service({
    url: '/bbPiPerson/getBbPiPersonList',
    method: 'get',
    params
  })
}


// @Tags BbPiPerson
// @Summary 快速编辑
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.BbPiPerson true "快速编辑"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiPerson/quickEdit [post]
export const quickEdit = (data) => {
  return service({
    url: '/bbPiPerson/quickEdit',
    method: 'post',
    data
  })
}

// @Tags BbPiPerson
// @Summary 导出excelBbPiPerson列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页导出excelBbPiPerson列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /bbPiPerson/getBbPiPersonexcelList [get]
export const excelList = (params) => {
  return service({
    url: '/bbPiPerson/excelList',
    method: 'get',
    params
  })
}

File: server/router/business/bb_pi_person.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type BbPiPersonRouter struct {
}

// InitBbPiPersonRouter 初始化 BbPiPerson 路由信息
func (s *BbPiPersonRouter) InitBbPiPersonRouter(Router *gin.RouterGroup) {
	router := Router.Group("bbPiPerson").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("bbPiPerson")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.BbPiPersonApi
	{
		router.POST("createBbPiPerson", apiV1.CreateBbPiPerson)   // 新建BbPiPerson
		router.DELETE("deleteBbPiPerson", apiV1.DeleteBbPiPerson) // 删除BbPiPerson
		router.DELETE("deleteBbPiPersonByIds", apiV1.DeleteBbPiPersonByIds) // 批量删除BbPiPerson
		router.PUT("updateBbPiPerson", apiV1.UpdateBbPiPerson)    // 更新BbPiPerson
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findBbPiPerson", apiV1.FindBbPiPerson)        // 根据ID获取BbPiPerson
		routerNoRecord.GET("getBbPiPersonList", apiV1.GetBbPiPersonList)  // 获取BbPiPerson列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel BbPiPerson列表
	}
}
